lib/storage/src/content_manager/toc.rs

```diff
@@ -541,7 +542,17 @@ impl TableOfContent {
             removed.before_drop().await;
             let path = self.get_collection_path(collection_name);
             drop(removed);
-            remove_dir_all(path).map_err(|err| {
+
+            // Move collection to ".deleted" folder to prevent accidental reuse
```

Comment:
Do we need a test for this untrivial logic?

Reply:
I think it is tested with existing tests enough

Reply:
okay